akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -132,15 +130,16 @@ private[akkastream] class DocumentStage[T](
         case Failure(reason) => onFailure(reason)
 
         case Success(resp @ Some(r)) => {
+          val bulkIter = cursor.documentIterator(r).take(maxDocs - r.reply.startingFrom)
```

Comment:
Declaring the iterate even if there is nothing returned doesn't seem a good idea